tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/ILogger/DirectSubmission/Formatting/LoggerLogFormatter.cs

```diff
@@ -0,0 +1,160 @@
+            {
+                var writerWrapper = new WriterWrapper(writer);
+                scopeProvider.ForEachScope(
+                    (scope, wrapper) =>
+                    {
+                        // Add dictionary scopes to a properties object
+                        // TODO: We _should_ handle "duplicate" values, where the property has been added as state,
+                        // And then we try and add it again, but I don't know if that's entirely necessary
+                        if (scope is IEnumerable<KeyValuePair<string, object>> scopeItems)
+                        {
+                            foreach (var item in scopeItems)
+                            {
+                                var key = item.Key;
+                                haveSource |= LogFormatter.IsSourceProperty(key);
+                                haveService |= LogFormatter.IsServiceProperty(key);
+                                haveHost |= LogFormatter.IsHostProperty(key);
+                                haveTags |= LogFormatter.IsTagsProperty(key);
+                                haveEnv |= LogFormatter.IsEnvProperty(key);
+                                haveVersion |= LogFormatter.IsVersionProperty(key);
+
+                                LogFormatter.WritePropertyName(writer, key);
+                                LogFormatter.WriteValue(wrapper.Writer, item.Value);
+                            }
+                        }
+                        else
+                        {
+                            wrapper.Values.Add(scope); // add to list for inclusion in scope array
+                        }
+                    },
+                    writerWrapper);
+
+                if (writerWrapper.Values.Count > 0)
+                {
+                    writer.WritePropertyName("Scopes", escape: false);
+                    writer.WriteStartArray();
+
+                    for (var i = 0; i < writerWrapper.Values.Count; i++)
+                    {
+                        LogFormatter.WriteValue(writer, writerWrapper.Values[i]);
+                    }
+
+                    writer.WriteEndArray();
+                }
+            }
+
+            return new LogPropertyRenderingDetails(haveSource, haveService, haveHost, haveTags, haveEnv, haveVersion, messageTemplate);
+        }
+
+        private static string GetLogLevelString(int logLevel) =>
+            logLevel switch
+            {
+                0 => DirectSubmissionLogLevelExtensions.Verbose, // Trace
+                1 => DirectSubmissionLogLevelExtensions.Debug,
+                2 => DirectSubmissionLogLevelExtensions.Information,
+                3 => DirectSubmissionLogLevelExtensions.Warning,
+                4 => DirectSubmissionLogLevelExtensions.Error,
+                5 => DirectSubmissionLogLevelExtensions.Fatal, // Critical
+                _ => throw new ArgumentOutOfRangeException(nameof(logLevel))
```

Comment:
I commented in the test, but I don't think we should throw in here.

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1945/commits/7f655a451f63c85b0c2f961e65dd04892e996920